src/drivers/cmakeDriver.ts

```diff
@@ -1730,6 +1747,67 @@ export abstract class CMakeDriver implements vscode.Disposable {
         }
     }
 
+    private async _showTaskQuickPick(tasks: vscode.Task[]): Promise<vscode.Task | undefined> {
+        interface TaskItem extends vscode.QuickPickItem {
+            task: vscode.Task;
+        }
+        const choices = tasks.map((t): TaskItem => ({ label: t.name, task: t }));
+        const sel = await vscode.window.showQuickPick(choices, { placeHolder: localize('select.build.task', 'Select build task') });
+        return sel ? sel.task : undefined;
+    }
+
+    private async _findBuildTask(targets?: string[]): Promise<vscode.Task | undefined> {
+
+        // Fetch all CMake task
+        const tasks = (await vscode.tasks.fetchTasks({ type: 'cmake' })).filter(task => ((task.group?.id === vscode.TaskGroup.Build.id)));
```

Comment:
Prior to the existence of `cmake` type tasks, people had to use `shell`.  Should these be included in the search?

Reply:
You should probably filter out tasks that don't use presets that match the current active ones either.  Those will fail if run.

Reply:
> Prior to the existence of `cmake` type tasks, people had to use `shell`. Should these be included in the search?

No sure. Shell task could be created for many other reasons not related to cmake extension.
I believe that if one sets config.useBuildTask, one does it with intention to use cmake tasks.
Additionally from moment when cmake extension inttoduced its own task type, it should be depricated to use shell tas for cmake builds :)

Reply:
> You should probably filter out tasks that don't use presets that match the current active ones either. Those will fail if run.

I added also possibility to filter by preset (if build preset is set).